diff --git a/CMakeLists.txt b/CMakeLists.txt
index 29a8993..6a4fd23 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -108,7 +108,7 @@ include(GNUInstallDirs)
 # Build DOCUMENTATION (not in ALL target and only if Doxygen is found)
 option(BUILD_DOC "Build the HTML documentation (with doxygen if available)." OFF)
 
-set(OPENJPEG_INSTALL_SUBDIR "${PROJECT_NAME}-${OPENJPEG_VERSION_MAJOR}.${OPENJPEG_VERSION_MINOR}")
+set(OPENJPEG_INSTALL_SUBDIR "${PROJECT_NAME}")
 
 if(NOT OPENJPEG_INSTALL_JNI_DIR)
   if(WIN32)
