diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1b38ba4..a7806c8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,11 +43,12 @@ project(
   leptonica
   LANGUAGES C
   VERSION 1.85.0)
-set(CMAKE_C_STANDARD 11)
+set(CMAKE_C_STANDARD 99)
 set(CMAKE_C_STANDARD_REQUIRED ON)
 
 # Standard installation paths
 include(GNUInstallDirs)
+add_definitions(-DNO_CONSOLE_IO)
 
 set(MINIMUM_WEBPMUX_VERSION 0.5.0)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 55007ff..5493723 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -11,7 +11,7 @@
 file(GLOB hdr "*.h")
 file(GLOB src "*.c")
 
-string(REPLACE "-O3" "-O2" CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
+
 
 add_library                     (leptonica ${src} ${hdr})
 set_target_properties           (leptonica PROPERTIES VERSION   6.0.0)
